app/src/main/java/com/example/sobok_android/presentation/view/pill/add/PillAddBottomSheetFragment.kt

```diff
@@ -33,26 +30,46 @@ class PillAddBottomSheetFragment : BottomSheetDialogFragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.clPillAddMyPill.setOnClickListener {
-            _isMyPill = true
-            getPillCount()
-            val intent = Intent(activity, PillAddActivity::class.java)
-            intent.putExtra("isMyPill", true)
-            startActivity(intent)
+        observePillCount()
+        navigateToAddMyPill()
+        navigateToSendPill()
+        closeBottomSheet()
+    }
+
+    private fun observePillCount() {
+        mainViewModel.pillCount.observe(viewLifecycleOwner) {
+            _pillCount = it
         }
+    }
+
+    private fun closeBottomSheet() {
+        binding.ivCancel.setOnClickListener {
+            dismiss()
+        }
+    }
+
+    private fun navigateToSendPill() {
         binding.clPillAddSendPill.setOnClickListener {
-            _isMyPill = false
-            getPillCount()
             val intent = Intent(activity, PillAddActivity::class.java)
             intent.putExtra("isMyPill", false)
             startActivity(intent)
         }
-        binding.ivCancel.setOnClickListener {
-            dismiss()
+    }
+
+    private fun navigateToAddMyPill() {
+        binding.clPillAddMyPill.setOnClickListener {
+            getPillCount()
         }
     }
 
     private fun getPillCount() {
         mainViewModel.getPillCount()
+        mainViewModel._pillCount.observe(this) {
+            _pillCount = it
+            val intent = Intent(activity, PillAddActivity::class.java)
+            intent.putExtra("isMyPill", true)
+            intent.putExtra("pillCount", _pillCount)
+            startActivity(intent)
+        }
```

Comment:
별거 아니긴 한데 이렇게 쓸 수도 있습니다!
```suggestion
        mainViewModel._pillCount.observe(this) {
            _pillCount = it
            startActivity(Intent(this, PillAddActivity::class.java).apply {
                .putExtra("isMyPill", true)
                .putExtra("pillCount", _pillCount)
            })
        }
```

Reply:
음.. 그리고 pillCount변수에 언더바가 붙어있는거 보니까 private 변수가 아닌거 같은데 요거는 나중에 수정해주세요!

Reply:
우왕 신기하다! 수정했슴다